lib/src/isolate_connection_factory.dart

```diff
@@ -103,6 +103,6 @@ class _IsolateSqliteConnection extends SqliteConnectionImpl {
   @override
   Future<void> close() async {
     await super.close();
-    closeFunction();
+    await closeFunction();
```

Comment:
Not awaiting here can cause the Isolate to be closed in the middle of the Mutex close process, resulting in a lock never released. Properly handling that case automatically is tricky, so for now make sure to always await closing in an Isolate.